web_infrastructure/nginx_status_facts.py

```diff
@@ -0,0 +1,144 @@
+description:
+  - Gathers facts from nginx from an URL having C(stub_status) enabled.
+version_added: "2.3"
+author: "René Moser (@resmo)"
+options:
+  url:
+    description:
+      - URL of the nginx status.
+    required: true
+requirements:
+  - "nginx > 1.3.14"
+
+notes:
+  - See http://nginx.org/en/docs/http/ngx_http_stub_status_module.html for more information.
+'''
+
+EXAMPLES = '''
+# Gather status facts from nginx on localhost
+- name: get current http stats
+  nginx_status_facts: url=http://localhost/nginx_status
+'''
+
+RETURN = '''
+---
+nginx_status_facts.active_connections:
+  description: Active connections.
+  returned: success
+  type: int
+  sample: 2340
+nginx_status_facts.accepts:
+  description: The total number of accepted client connections.
+  returned: success
+  type: int
+  sample: 81769947
+nginx_status_facts.handled:
+  description: The total number of handled connections. Generally, the parameter value is the same as accepts unless some resource limits have been reached.
+  returned: success
+  type: int
+  sample: 81769947
+nginx_status_facts.requests:
+  description: The total number of client requests.
+  returned: success
+  type: int
+  sample: 144332345
+nginx_status_facts.reading:
+  description: The current number of connections where nginx is reading the request header.
+  returned: success
+  type: int
+  sample: 0
+nginx_status_facts.writing:
+  description: The current number of connections where nginx is writing the response back to the client.
+  returned: success
+  type: int
+  sample: 241
+nginx_status_facts.waiting:
+  description: The current number of idle client connections waiting for a request.
+  returned: success
+  type: int
+  sample: 2092
+nginx_status_facts.data:
+  description: HTTP response as is.
+  returned: success
+  type: string
+  sample: "Active connections: 2340 \nserver accepts handled requests\n 81769947 81769947 144332345 \nReading: 0 Writing: 241 Waiting: 2092 \n"
+'''
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import fetch_url
+
+
+class NginxStatusFacts(object):
+
+    def __init__(self):
+        self.url = module.params.get('url')
+
+    def run(self):
+        result = {
+            'nginx_status_facts': {
+                'active_connections': None,
+                'accepts': None,
+                'handled': None,
+                'requests': None,
+                'reading': None,
+                'writing': None,
+                'waiting': None,
+                'data': None,
+            }
+        }
+        (response, info) = fetch_url(module, self.url , force=True)
+        if response:
+            data = response.read()
+            if not data:
+                return result
+            result['nginx_status_facts']['data'] = data
+            match = re.match(r'Active connections: ([0-9]+) \nserver accepts handled requests\n ([0-9]+) ([0-9]+) ([0-9]+) \nReading: ([0-9]+) Writing: ([0-9]+) Waiting: ([0-9]+)', data, re.S)
```

Comment:
I do not know nginx, so the question may seems silly, is there any risk of this string being translated in the future ?